exporter.go

```diff
@@ -371,16 +371,28 @@ func (e *Exporter) entrypoint(launchMD launch.Metadata, defaultProcessType strin
 	if !e.supportsMulticallLauncher() {
 		return launch.LauncherPath, nil
 	}
-	if defaultProcessType == "" {
-		if len(launchMD.Processes) == 1 {
-			e.Logger.Infof("Setting default process type '%s'", launchMD.Processes[0].Type)
-			return launch.ProcessPath(launchMD.Processes[0].Type), nil
+
+	if defaultProcessType == "" && e.PlatformAPI.Compare(api.MustParse("0.6")) < 0 && len(launchMD.Processes) == 1 {
+		// if there is only one process, we set it to the default for platform API < 0.6
+		e.Logger.Infof("Setting default process type '%s'", launchMD.Processes[0].Type)
+		return launch.ProcessPath(launchMD.Processes[0].Type), nil
+	}
+
+	if defaultProcessType != "" {
+		defaultProcess, ok := launchMD.FindProcessType(defaultProcessType)
```

Comment:
The `launchMD.FindProcessType` finds the first matching process declared in the metadata with the matching type. Is the order of `launchMD.Processes` going to shift at all to make this unexpected for users on `< 0.6`?

I don't _think_ there is anything to worry about here, but I wanted to ask just in case. It looks innocent but if we are storing multiple processes of the same type and have changed that order we may write the incorrect process.

Reply:
I think the previous use of `procMap` ensured that we only ever had one process of a given type.

Reply:
That's right. We used `procMap` in the `builder` to keep only one process for each type, and now we're doing it with the new struct `orderedProcesses`.
This is why we need the `map` in this struct.